community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/OnlineIndexProxy.java

```diff
@@ -46,14 +46,42 @@
     private final IndexConfiguration configuration;
     private final IndexCountsRemover indexCountsRemover;
 
+    // About this flag: there are two online "modes", you might say...
```

Comment:
would it be better if we had a separate applier instead that release lock group in the end of the batch?